File: Problems/No.89_Gray_Code/gray_code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        
        // parameters
        std::vector<int> GCode = {0, 1};
        int cNum;
        
        // start computing
        for (int i=2; i<(1<<n); i++) {
            
            cNum = i ^ (i>>1);
            GCode.push_back(cNum);

        }

        return GCode;

    }
};

File: Problems/No.89_Gray_Code/gray_code.py
class Solution(object):
    def grayCode(self, n):
        """
        :type n: int
        :rtype: List[int]
        """

        # parameters
        GCode = [0, 1]
        cNum = None

        # start computing
        for i in range(2, (1<<n)):
            cNum = i ^ (i>>1)
            GCode.append(cNum)

        return GCode
